code/build/template/include/extensions/[+python orxPy.h]

```diff
@@ -0,0 +1,2613 @@
+          py_newstr(py_offset(pyArgv, 1), orxBody_GetPartName(pstPayload->pstSenderPart));
+          py_neworxobject(py_offset(pyArgv, 2), (orxOBJECT *)_pstEvent->hRecipient);
+          py_newstr(py_offset(pyArgv, 3), orxBody_GetPartName(pstPayload->pstRecipientPart));
+          // Call handler
+          orxPy_CallN(orxPy_GetCallback(sstPy.pyOnSeparate), s32Argc, pyArgv);
+        }
+        break;
+      }
+      default:
+      {
+        orxLOG(orxPY_KZ_LOG_TAG "orxPy_EventHandler: Unsupported physics event %d", _pstEvent->eID);
+        orxASSERT(orxFALSE);
+        break;
+      }
+      }
+      orxPROFILER_POP_MARKER();
+      break;
+    }
+    case orxEVENT_TYPE_SHADER:
+    {
+      orxPROFILER_PUSH_MARKER("orxPy_EventHandler_Shader");
+      orxASSERT(_pstEvent->eID == orxSHADER_EVENT_SET_PARAM);
+      orxSHADER_EVENT_PAYLOAD *pstPayload = (orxSHADER_EVENT_PAYLOAD *)_pstEvent->pstPayload;
+      if (sstPy.pyOnShaderParam != 0 && (pstPayload->eParamType == orxSHADER_PARAM_TYPE_FLOAT || pstPayload->eParamType == orxSHADER_PARAM_TYPE_VECTOR))
+      {
+        // Build argument list
+        orxS32 s32Argc = 4;
+        py_Ref pyArgv = py_r0();
+        py_newlistn(pyArgv, s32Argc);
+        py_neworxobject(py_offset(pyArgv, 0), (orxOBJECT *)_pstEvent->hSender);
+        py_newstr(py_offset(pyArgv, 1), pstPayload->zShaderName);
+        py_newstr(py_offset(pyArgv, 2), pstPayload->zParamName);
+        py_assign(py_offset(pyArgv, 3), pstPayload->eParamType == orxSHADER_PARAM_TYPE_FLOAT ? py_tpobject(tp_float) : py_tpobject(tp_orxvector));
+
+        // Call handler
+        orxSTATUS eShaderResult = orxPy_CallN(orxPy_GetCallback(sstPy.pyOnShaderParam), s32Argc, pyArgv);
+        if (eShaderResult == orxSTATUS_SUCCESS && (py_isfloat(py_retval()) || py_isint(py_retval()) || py_istype(py_retval(), tp_orxvector)))
+        {
+          switch (pstPayload->eParamType)
+          {
+          case orxSHADER_PARAM_TYPE_FLOAT:
+          {
+            if (py_isfloat(py_retval()) || py_isint(py_retval()))
+            {
+              py_castfloat32(py_retval(), &pstPayload->fValue);
+            }
+            break;
+          }
+          case orxSHADER_PARAM_TYPE_VECTOR:
+          {
+            if (py_istype(py_retval(), tp_orxvector))
+            {
+              orxVECTOR *pvVal = (orxVECTOR *)py_touserdata(py_retval());
+              orxVector_Copy(&pstPayload->vValue, pvVal);
+            }
+            break;
+          }
+          default:
+          {
+            // An unsupported parameter type
+            break;
+          }
+          }
+        }
+      }
+      orxPROFILER_POP_MARKER();
+      break;
+    }
+    default:
+    {
+      orxLOG(orxPY_KZ_LOG_TAG "orxPy_EventHandler: Unsupported event (%d, %d)", _pstEvent->eType, _pstEvent->eID);
+      orxASSERT(orxFALSE);
+      break;
+    }
+    }
+
+    orxPROFILER_POP_MARKER();
+
+    // Done!
+    return eResult;
+  }
+
+  orxSTATUS orxPy_InitVM()

```

Comment:
Initialize the pocketpy virtual machine. This does not execute any game-specific Python code. It does include all of the `orx.*` engine module bindings.